src/cli_config.rs

```diff
@@ -0,0 +1,83 @@
+use std::net::ToSocketAddrs;
+
+use clap::{Arg, App, AppSettings};
+use hex::FromHex;
+use itertools::Itertools;
+use num_cpus;
+use tox::toxcore::crypto_core::*;
+use tox::toxcore::dht::packed_node::PackedNode;
+
+/// Config parsed from command line arguments.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct CliConfig {
+    /// List of bootstrap nodes.
+    pub bootstrap_nodes: Vec<PackedNode>,
+    /// Number of threads for execution. None if single threaded runtime should
+    /// be used.
+    pub threads_count: Option<usize>,
+}
+
+/// Parse command line arguments.
+pub fn cli_parse() -> CliConfig {
+    let num_cpus_string = num_cpus::get().to_string();
```

Comment:
See https://docs.rs/tokio/0.1.7/tokio/executor/thread_pool/struct.Builder.html#method.pool_size .

Reply:
You don't need to compute the number of threads. It is computed if you don't set it up calling method `.pool_size`.

See basic usage: https://github.com/tokio-rs/tokio/blob/master/tests/runtime.rs#L58..L77

Reply:
Then I won't be able to specify default value for clap here https://github.com/tox-rs/tox-node/pull/4/files#diff-d1d758c962e4017c7e616dc7d009ff43R50 and will need to handle this case somehow.

Reply:
Please take a look at this [article](https://coaxion.net/blog/2018/04/improving-gstreamer-performance-on-a-high-number-of-network-streams-by-sharing-threads-between-elements-with-rusts-tokio-crate/). The main idea: single-threaded runtime is the best option for a network application.

And we must provide the best option 1) by default 2) allow the users pass it as an explicit option like `-j=-1` for nerds. Your way of dealing with args does none of these requirements.

Reply:
By default it's singlethreaded. Why somebody needs to specify this as `-j=-1`?

Reply:
Like: `alias st_node='tox-node -j1'`, where st=single threaded. Since you don't configure this option as `.multiple(true)` the second `-jN` is forbidden.